.github/workflows/format.yaml

```diff
@@ -0,0 +1,40 @@
+# This is a format job. Pre-commit has a first-party GitHub action, so we use
+# that: https://github.com/pre-commit/action
+
+name: Format
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
```

Comment:
@zultron can you add make it so it will only run on main push.

Reply:
Sorry @vatanaksoytezer , can you look at #3 for this update?